### PR TITLE
CF-112 add dropdown to categorise courses into different subgroups

### PR DESCRIPTION
**Pull request by gtangelo**
Allow the user to categorises courses into different subgroups (i.e. Major - Core Courses, Minor - Prescribed Electives, etc.) via the dropdown next to the add to planner button

Example in some cases below:
image
image
image

This is then categorise in the Term Planner as well depending on what the user selects in the dropdown
image

### CODE REVIEW

File: frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx [hunk 1]
```diff
@@ -23,7 +23,46 @@ const CourseAttribute = ({ title, content }) => {
...
 };
 
-export default function CourseDescription() {
+const PlannerDropdown = ({courseCode, structure, addToPlanner}) => {
+  const [categories, setCategories] = useState([])
+
+  useEffect(() => {
+    // Example groups: Major, Minor, General
+    const categoriesDropdown = []
+    for (const group in structure) {
```
Comment (Peedee2002) [inline frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx:32]: we can leave it like this for now - but we might need to expand this to have an option to "double count"

File: frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
```
Comment (Peedee2002) [inline frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx:1]: nice

File: frontend/src/pages/CourseSelector/main.jsx [hunk 3]
```diff
@@ -7,23 +8,47 @@ import "./main.less";
...
+  React.useEffect(() => {
+    fetchStructure();
+  }, []);
+
+   // get structure of degree
+   const fetchStructure = async () => {
+    try {
+      const res1 = await axios.get(
+        `http://localhost:8000/programs/getStructure/${programCode}/${specialisation}/${
+          minor !== "" ? minor : ""
```
Comment (Peedee2002) [inline frontend/src/pages/CourseSelector/main.jsx]: this is redundant, right?

Comment (gtangelo) [reply frontend/src/pages/CourseSelector/main.jsx]: yep. Just fixed that!

**Review by Snuzzn — APPROVED** [review]
lgtm! Your comments are very helpful :)

**Pull request merged**